csharp/extractor/Semmle.Extraction.CSharp.Standalone/AssemblyCache.cs

```diff
@@ -165,32 +163,32 @@ public AssemblyInfo ResolveReference(string id)
         /// <returns>The assembly info.</returns>
         public AssemblyInfo GetAssemblyInfo(string filepath)
         {
-            if(assemblyInfo.TryGetValue(filepath, out var info))
+            if (assemblyInfoByFileName.TryGetValue(filepath, out var info))
             {
                 return info;
             }
-            else
+
+            IndexReference(filepath);
+
+            if (assemblyInfoByFileName.TryGetValue(filepath, out info))
             {
-                info = AssemblyInfo.ReadFromFile(filepath);
-                assemblyInfo.Add(filepath, info);
                 return info;
             }
+
+            return AssemblyInfo.Invalid;
         }
 
-        // List of pending DLLs to index.
-        readonly List<string> dlls = new List<string>();
+        readonly Queue<string> pendingDllsToIndex = new Queue<string>();
 
-        // Map from filename to assembly info.
-        readonly Dictionary<string, AssemblyInfo> assemblyInfo = new Dictionary<string, AssemblyInfo>();
+        readonly Dictionary<string, AssemblyInfo> assemblyInfoByFileName = new Dictionary<string, AssemblyInfo>();
```

Comment:
I'm not entirely sure if we need these duplicate dictionaries and lists "by filename" and "by id".